internal/controller/rabbitmq/rabbitmqvhost_controller.go

```
@@ -0,0 +1,233 @@
+	// If protection finalizer exists, check if still actively used
+	if controllerutil.ContainsFinalizer(instance, rabbitmqv1.VhostFinalizer) {
+		// Check if still actively used by TransportURL
+		for _, owner := range instance.OwnerReferences {
+			if owner.Controller != nil && *owner.Controller && owner.Kind == "TransportURL" {
+				transportURL := &rabbitmqv1.TransportURL{}
+				if err := r.Get(ctx, types.NamespacedName{Name: owner.Name, Namespace: instance.Namespace}, transportURL); err == nil && transportURL.DeletionTimestamp.IsZero() {
+					vhost := transportURL.Spec.Vhost
+					if vhost == "" {
+						vhost = "/"
+					}
+					expectedRef := ""
+					if vhost != "/" {
+						expectedRef = fmt.Sprintf("%s-%s-vhost", transportURL.Name, vhost)
+					}
+					if expectedRef == instance.Name {
+						return ctrl.Result{RequeueAfter: time.Duration(2) * time.Second}, nil
+					}
+				}
+			}
+		}
+		// Check if any active users reference this vhost
+		userList := &rabbitmqv1.RabbitMQUserList{}
+		if err := r.List(ctx, userList, client.InNamespace(instance.Namespace)); err == nil {
+			for _, user := range userList.Items {
+				if user.Spec.VhostRef == instance.Name && user.DeletionTimestamp.IsZero() {
+					return ctrl.Result{RequeueAfter: time.Duration(2) * time.Second}, nil
+				}
+			}
+		}
+		controllerutil.RemoveFinalizer(instance, rabbitmqv1.VhostFinalizer)
+	}
+
+	// Get RabbitMQ cluster
+	rabbit := &rabbitmqclusterv2.RabbitmqCluster{}
+	err := r.Get(ctx, types.NamespacedName{Name: instance.Spec.RabbitmqClusterName, Namespace: instance.Namespace}, rabbit)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	if err == nil {
+		// Get admin credentials
+		rabbitSecret, _, err := oko_secret.GetSecret(ctx, h, rabbit.Status.DefaultUser.SecretReference.Name, instance.Namespace)
+		if err == nil {
```

can we ignore err!=nil , which are not IsNotFound(err)?